node.js

```diff
@@ -663,160 +869,239 @@ async function stabilize() {
-        return false;
+        // TODO try to rescue it by stepping through the rest of its finger table, else destroy it
+        predecessor_seems_ok = false;
+        return predecessor_seems_ok;
     }
     if (predecessor.id !== _self.id) {
-        other_node_client = caller(`localhost:${predecessor.port}`, PROTO_PATH, "Node");
         try {
             // confirm that the predecessor is actually there
-            await other_node_client.getPredecessor(_self);
-            // then kick by setting the successor to the same as the predecessor
-            FingerTable[0].successor = predecessor;
+            predecessor_seems_ok = await check_predecessor();
         } catch (err) {
+            predecessor_seems_ok = false;
             console.error(err);
         }
+        if (predecessor_seems_ok) {
+            // then kick by setting the successor to the same as the predecessor
+            FingerTable[0].successor = predecessor;
+            SuccessorTable[0] = FingerTable[0].successor;
+        }
     } else {
         console.log("\nWarning: {", _self.id, "} is isolated because",
             "predecessor is", predecessor.id,
             "and successor is", FingerTable[0].successor.id, ".");
-        return false;
+            predecessor_seems_ok = true;
     }
-    return true;
+    return predecessor_seems_ok;
 }
 
-/* modified 20191021 */
+/**
+ * Directly implements the pseudocode's notify() method.
+ * 
+ * @version 20191021
+ */
 async function notify(message, callback) {
-    /**
-     * Directly implements the pseudocode's notify() method.
-     */
     const n_prime = message.request;
     // if (predecessor is nil or n' 'is in' (predecessor, n))
-    if ( (predecessor.id == null)
-        || isInModuloRange(n_prime.id, predecessor.id, false, _self.id, false) ) {
+    if ((predecessor.id == null)
+        || isInModuloRange(n_prime.id, predecessor.id, false, _self.id, false)) {
         // predecessor = n';
         predecessor = n_prime;
     }
     callback(null, {});
 }
 
-/* modified 20191022 */
+/**
+ * Directly implements the pseudocode's fix_fingers() method.
+ * 
+ * @version 20191103
+ */
 async function fix_fingers() {
-    /**
-     * Directly implements the pseudocode's fix_fingers() method.
-     */
     // enable debugging output
     const DEBUGGING_LOCAL = false;
+    let n_successor = NULL_NODE;
 
     // i = random index > 1 into finger[]; but really >0 because 0-based
     // random integer within the range (0, m)
     const i = Math.ceil(Math.random() * (HASH_BIT_LENGTH - 1));
     // finger[i].node = find_successor(finger[i].start);
-    FingerTable[i].successor = await find_successor(FingerTable[i].start, _self, _self);
+    try {
+        n_successor = await find_successor(FingerTable[i].start, _self, _self);
+        if (n_successor.id !== null) {
+            FingerTable[i].successor = n_successor;
+        }
+    } catch (err) {
+        // don't change the finger just yet
```

Comment:
Can you please log here? I don't like to see errors caught and silently suppressed.